freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/config/FreeIpaConfigView.java

```diff
@@ -0,0 +1,248 @@
+
+    private final String realm;
+
+    private final String domain;
+
+    private final String password;
+
+    private final String reverseZones;
+
+    private final String adminUser;
+
+    private final String freeipaToReplicate;
+
+    private final Set<Object> hosts;
+
+    private final String azureInstanceMsi;
+
+    @SuppressWarnings("ExecutableStatementCount")
+    private FreeIpaConfigView(Builder builder) {
+        this.backupEnabled = builder.backupEnabled;
+        this.monthlyFullBackUpEnabled = builder.monthlyFullBackUpEnabled;
+        this.hourlyBackUpEnabled = builder.hourlyBackUpEnabled;
+        this.initialFullBackupEnabled = builder.initialFullBackupEnabled;
+        this.backupLocation = builder.backupLocation;
+        this.backupPlatform = builder.backupPlatform;
+        this.realm = builder.realm;
+        this.domain = builder.domain;
+        this.password = builder.password;
+        this.reverseZones = builder.reverseZones;
+        this.adminUser = builder.adminUser;
+        this.freeipaToReplicate = builder.freeipaToReplicate;
+        this.hosts = builder.hosts;
+        this.azureInstanceMsi = builder.azureInstanceMsi;
+    }
+
+    public String getBackupLocation() {
+        return backupLocation;
+    }
+
+    public String getRealm() {
+        return realm;
+    }
+
+    public String getDomain() {
+        return domain;
+    }
+
+    public String getPassword() {
+        return password;
+    }
+
+    public String getReverseZones() {
+        return reverseZones;
+    }
+
+    public String getAdminUser() {
+        return adminUser;
+    }
+
+    public String getFreeipaToReplicate() {
+        return freeipaToReplicate;
+    }
+
+    public Set<Object> getHosts() {
+        return hosts;
+    }
+
+    public boolean isBackupEnabled() {
+        return backupEnabled;
+    }
+
+    public boolean isMonthlyFullBackUpEnabled() {
+        return monthlyFullBackUpEnabled;
+    }
+
+    public boolean isHourlyBackUpEnabled() {
+        return hourlyBackUpEnabled;
+    }
+
+    public boolean isInitialFullBackupEnabled() {
+        return initialFullBackupEnabled;
+    }
+
+    public String getBackupPlatform() {
+        return backupPlatform;
+    }
+
+    public String getAzureInstanceMsi() {
+        return azureInstanceMsi;
+    }
+
+    @SuppressWarnings("ExecutableStatementCount")
+    public Map<String, Object> toMap() {
+        Map<String, Object> map = new HashMap<>();
+        map.put("backup_enabled", this.backupEnabled);
+        map.put("backup_location", ObjectUtils.defaultIfNull(this.backupLocation, EMPTY_CONFIG_DEFAULT));
+        map.put("monthly_full_backup_enabled", this.monthlyFullBackUpEnabled);
+        map.put("hourly_backup_enabled", this.hourlyBackUpEnabled);
+        map.put("initial_full_backup_enabled", this.initialFullBackupEnabled);
+        map.put("backup_platform", ObjectUtils.defaultIfNull(this.backupPlatform, EMPTY_CONFIG_DEFAULT));
```

Comment:
backup related stuff could be under backup as they logically stick together

Reply:
Ill see if I can do this

Reply:
Changed the structure